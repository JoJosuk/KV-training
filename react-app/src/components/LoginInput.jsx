import { forwardRef } from "react";
import React from "react";
const WrapperLoginInput = (
  {
    type,
    labelContent,
    inputPlaceholder,
    name,
    id,
    value = "",
    error = false,
    onValueChange = () => null,
  },
  ref
) => {
  return (
    <>
      <span className={`${error ? "red" : ""}`}>
        <label htmlFor={id}> {labelContent}</label>
        <input
          data-testid="input-container"
          ref={ref}
          type={type}
          name={name}
          id={id}
          placeholder={inputPlaceholder}
          value={value}
          onChange={onValueChange}
        />
      </span>
    </>
  );
};

const LoginInput = forwardRef(WrapperLoginInput);
export default LoginInput;
