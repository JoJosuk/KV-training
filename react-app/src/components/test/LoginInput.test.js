import React from "react";
import { fireEvent, render } from "@testing-library/react";
import LoginInput from "../LoginInput";
describe("check if input works properly", () => {
  test("check if Input renders correctly ", () => {
    const { getByTestId } = render(<LoginInput />);
    const element = getByTestId("input-container");
    expect(element).toBeTruthy();
  });
  test("check if Placeholder works properly ", () => {
    const { getByPlaceholderText } = render(
      <LoginInput inputPlaceholder={"hello"} />
    );
    getByPlaceholderText("hello");
  });

  test("Check if label is working properly", () => {
    const { getByLabelText } = render(
      <LoginInput labelContent={"test"} id={"testid"} />
    );
    getByLabelText("test");
  });

  test("check if value is working properly", () => {
    const onChange = jest.fn();
    const { getByDisplayValue } = render(
      <LoginInput id={"testid"} value="hello" />
    );
    const element = getByDisplayValue("hello");
    expect(element).toBeTruthy();
  });
  test("check if value is changing properly", () => {
    let letval = "";
    const onChange = jest.fn((e) => {
      letval = e.target.value;
    });
    const { getByTestId } = render(
      <LoginInput id={"testid"} value={letval} onValueChange={onChange} />
    );
    const element = getByTestId("input-container");
    fireEvent.change(element, { target: { value: "new value" } });

    expect(letval).toBe("new value");
  });
  test("check snapshot is matched properly", () => {
    const onValueChange = jest.fn();
    const { asFragment } = render(
      <LoginInput
        id={"testid"}
        inputPlaceholder={"placeholder"}
        labelContent={"label"}
        name={"name"}
        type={"text"}
        value={"value"}
        onValueChange={onValueChange}
      />
    );
    expect(asFragment).toMatchSnapshot();
  });
});
