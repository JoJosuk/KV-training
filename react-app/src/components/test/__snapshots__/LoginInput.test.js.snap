// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if input works properly check snapshot is matched properly 1`] = `[Function]`;
